code/datums/ammo/misc.dm

```diff
@@ -174,6 +174,84 @@
 		var/obj/item/weapon/gun/flare/flare_gun_fired_from = fired_projectile.shot_from
 		flare_gun_fired_from.last_signal_flare_name = signal_flare.name
 
+/datum/ammo/arrow
+	name = "arrow"
+	ping = null //no bounce off.
+	damage_type = BRUTE
+	icon_state = "arrow"
+
+	damage = 110
+	penetration = 20
+	accuracy = HIT_ACCURACY_TIER_3
+	max_range = 14
+	shell_speed = AMMO_SPEED_TIER_3
+	flags_ammo_behavior = AMMO_SPECIAL_EMBED
+	shrapnel_chance = SHRAPNEL_CHANCE_TIER_10
+	shrapnel_type = /obj/item/arrow
+	handful_type = /obj/item/arrow
+	sound_hit = 'sound/weapons/pierce.ogg'
+	var/activated = FALSE
+
+/datum/ammo/arrow/on_embed(mob/embedded_mob, obj/limb/target_organ, silent = FALSE)
+	if(ishuman(embedded_mob) && !isyautja(embedded_mob))
+		if(istype(target_organ))
+			target_organ.embed(new shrapnel_type)
+
+/datum/ammo/arrow/proc/drop_arrow(turf/T, obj/projectile/fired_projectile)
+	var/obj/item/arrow/G = new handful_type(T)
+	var/matrix/rotation = matrix()
+	rotation.Turn(fired_projectile.angle - 90)
+	G.apply_transform(rotation)
+
+/datum/ammo/arrow/on_hit_mob(mob/mob,obj/projectile/projectile)
+	mob.apply_effect(1, STUN)
+	mob.apply_effect(3, DAZE)
+	pushback(mob, projectile, 7)
+
+/datum/ammo/arrow/on_hit_obj(obj/O,obj/projectile/P)
+	drop_arrow(get_turf(P), P)
+
+/datum/ammo/arrow/on_hit_turf(turf/T, obj/projectile/P)
+	if(T.density && isturf(P.loc))
+		drop_arrow(P.loc, P)
+	else
+		drop_arrow(T, P)
+
+/datum/ammo/arrow/do_at_max_range(obj/projectile/P, mob/firer)
+	drop_arrow(get_turf(P), P)
+
+/datum/ammo/arrow/expl
+	activated = TRUE
+	handful_type = /obj/item/arrow/expl
+	damage_type = BURN
+	flags_ammo_behavior = AMMO_HITS_TARGET_TURF
+	shrapnel_chance = 0
+	var/datum/effect_system/smoke_spread/smoke
```

Comment:
```suggestion
/datum/ammo/arrow/expl
	activated = TRUE
	handful_type = /obj/item/arrow/expl
	damage_type = BURN
	flags_ammo_behavior = AMMO_HITS_TARGET_TURF
	shrapnel_chance = 0
	var/datum/effect_system/smoke_spread/smoke

/datum/ammo/arrow/expl/New()
	. = ..()
	smoke = new()
```
Otherwise it's going to runtime.